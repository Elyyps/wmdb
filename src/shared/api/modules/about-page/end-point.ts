import { generateAboutPageDummyData } from "./generate-dummy-data";

export const generateAboutPageData = generateAboutPageDummyData;
