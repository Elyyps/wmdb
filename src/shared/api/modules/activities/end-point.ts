import { generateActivitiesDummyData } from "./dummy-data";

export const generateActivitiesData = generateActivitiesDummyData;
