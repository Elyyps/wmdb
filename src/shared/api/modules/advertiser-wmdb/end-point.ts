import { generateAdvertiserDummyData } from "./dummy-data";
export const generateAdvertiserData = generateAdvertiserDummyData;
