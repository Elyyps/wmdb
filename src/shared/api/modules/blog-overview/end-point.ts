import { generateBlogOverviewDummyData, generateCards } from "./dummy-data";
import { ICompanyOutingCard } from "@app/api/core/company-outing-card";

export const generateBlogOverviewData = generateBlogOverviewDummyData;
const amountOfCards = 50;
export const getCardsWithPagination = (skip: number, take: number): ICompanyOutingCard[] => {
  let cards: ICompanyOutingCard[] = generateCards(amountOfCards);
  cards = cards.slice(skip, take + skip);

  return cards;
};
