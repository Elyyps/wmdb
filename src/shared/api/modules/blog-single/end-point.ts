import { generateBlogSingleDummyData } from "./dummy-data";

export const generateBlogSingleData = generateBlogSingleDummyData;
