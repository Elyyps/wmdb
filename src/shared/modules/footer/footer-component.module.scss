.footer {
  width: 100%;
  font-size: 14px;
  font-weight: 600;
  background: $global-muted-background;
  padding: 24px 0;
  @media (min-width: $breakpoint-small) {
    padding: 32px 0;
  }

  &__holder {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    @media (min-width: $breakpoint-medium) {
      flex-direction: row;
      justify-content: space-between;
    }
  }

  &__logo {
    margin: 0 auto 16px;
    display: flex;
    justify-content: center;
    @media (min-width: $breakpoint-medium) {
      margin: 0;
    }
  }

  &__side {
    display: flex;
    flex-direction: column;
    align-items: center;
    a {
      display: flex;
    }
    @media (min-width: $breakpoint-medium) {
      flex-direction: row;
      display: flex;
    }
  }

  &__nav {
    display: grid;
    grid-template-columns: auto auto auto auto;
    margin: 0;
    padding: 0;

    @media (max-width: $breakpoint-medium) {
      display: flex;
      flex-direction: column;
    }

    li {
      font-size: 14px;
      margin-right: 30px;
      margin-bottom: 8px;
      @media (max-width: $breakpoint-medium) {
        margin: 0 auto 8px;
      }
    }
    a {
      color: $global-color;
      text-decoration: none;
      white-space: nowrap;

      &:hover {
        color: $brand-primary;
      }
    }
  }
}
