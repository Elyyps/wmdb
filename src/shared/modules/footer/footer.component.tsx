import * as React from "react";
import styles from "./footer-component.module.scss";
import { LogoComponent } from "@app/prep/modules-prep/core";
import { LinkComponent } from "@app/core/link";
import { IconComponent } from "@app/core/icon";
import Facebook from "@assets/icons/facebook.svg";
import { IFooter } from "@app/api/modules/footer/footer.module";

export interface IFooterComponentProps {
  footerModule: IFooter;
}

const FooterComponent = (props: IFooterComponentProps) => (
  <div className={styles["footer"]}>
    <div className="uk-container">
      <div className={styles["footer__holder"]}>
        <div className={styles["footer__logo"]}>
          <LogoComponent />
        </div>

        <div className={styles["footer__side"]}>
          <ul className={styles["footer__nav"]}>
            {props.footerModule.links.map((item, key) => (
              <li key={key}>
                <LinkComponent to={item.url}>{item.title}</LinkComponent>
              </li>
            ))}
          </ul>

          <LinkComponent to={"/"}>
            <IconComponent icon={Facebook} size="24px" fillColor={"#3b5998"} />
          </LinkComponent>
        </div>
      </div>
    </div>
  </div>
);

export { FooterComponent };
