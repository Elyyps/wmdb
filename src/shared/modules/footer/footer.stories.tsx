import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { FooterComponent } from "./footer.component";
import { generateFooterData } from "@app/api/modules/footer/end-point";

storiesOf("Footer", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <FooterComponent footerModule={generateFooterData()} />);
