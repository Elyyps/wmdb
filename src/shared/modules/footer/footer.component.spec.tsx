import React from "react";
import { shallow } from "enzyme";
import { FooterComponent } from "./footer.component";
import { generateFooterData } from "@app/api/modules/footer/end-point";

describe("[Footer]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<FooterComponent footer={generateFooterData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
