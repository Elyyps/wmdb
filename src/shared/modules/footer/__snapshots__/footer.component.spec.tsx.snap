// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Footer] should render component without crashing 1`] = `
<div
  className="footer"
>
  <div
    className="uk-container"
  >
    <div
      className="footer__holder"
    >
      <div
        className="footer__logo"
      >
        <LogoComponent
          logo="WMDB.svg"
        />
      </div>
      <div
        className="footer__side"
      >
        <ul
          className="footer__nav"
        >
          <li
            key="0"
          >
            <LinkComponent
              to="/"
            >
              info@wmdb.nl
            </LinkComponent>
          </li>
          <li
            key="1"
          >
            <LinkComponent
              to="/"
            >
              Web-Effects
            </LinkComponent>
          </li>
          <li
            key="2"
          >
            <LinkComponent
              to="/"
            >
              Algemene voorwaarden
            </LinkComponent>
          </li>
          <li
            key="3"
          >
            <LinkComponent
              to="/"
            >
              Privacystatement
            </LinkComponent>
          </li>
          <li
            key="4"
          >
            <LinkComponent
              to="/"
            >
              Disclaimer
            </LinkComponent>
          </li>
        </ul>
        <a
          href="/"
          role="button"
        >
          <IconComponent
            fillColor="#3b5998"
            icon="facebook.svg"
            size="24px"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
